### ISSUE TITLE
[feature] match host operating system

### ISSUE THREAD

**Issue by Merith-TK** [emeraldwalk/vscode-runonsave#85]
So I work on the same project with both windows and linux machines, the problem is that both have different command structures (linux uses $VAR and windows uses %VAR% for example)

I was hoping to be able to define an "OS LOCK" setting, where I can say "use this command for windows, and this command for linux"

**Comment by bmingles** [comment]
This is a good suggestion. Will consider how to add this in future release.

**Comment by Merith-TK** [comment]
Another reason for this request is that windows uses `\` in file paths, as well as the notorious `C:\Path` drive mounts, while in linux `\` is used as an escape character to allow for characters to be parsed as raw string and not format strings (commonly used in `VAR="echo \"example\""` while windows does not,

**Comment by bmingles** [comment]
Yes, I think this would be a good addition. I’ve marked it as a future enhancement

**Comment by Ben-Voris** [comment]
A variation on a theme is settings.json that is used in both Cygwin and WSL. In this case, the problem is just the location of the shell.
